hooks/cmd/install/install.go

```diff
@@ -0,0 +1,225 @@
+			}
+
+			srcP := fmt.Sprintf("%s/%s/configuration.toml", snapConfP, name)
+			sf, err := os.Open(srcP)
+			if err != nil {
+				return errors.New(fmt.Sprintf("Can't open %s - %v", srcP, err))
+			}
+			defer sf.Close()
+
+			destP := fmt.Sprintf("%s/configuration.toml", destD)
+			log.Info(fmt.Sprintf("edgex-asc:install creating: %s", destP))
+			df, err := os.OpenFile(destP, os.O_WRONLY|os.O_CREATE, 0755)
+			if err != nil {
+				return errors.New(fmt.Sprintf("Can't create %s - %v", destP, err))
+			}
+			defer df.Close()
+
+			_, err = io.Copy(df, sf)
+			if err != nil {
+				return errors.New(fmt.Sprintf("Can't copy %s to %s - %v", srcP, destP, err))
+			}
+		}
+	}
+
+	return nil
+}
+
+// TODO: merge w/configure version
+func handleVal(p string, k string, v interface{}, flatConf map[string]interface{}) {
+	var mk string
+
+	// top level keys don't include "env", so no separator needed
+	if p == "" {
+		mk = k
+	} else {
+		mk = fmt.Sprintf("%s.%s", p, k)
+	}
+
+	log.Info(fmt.Sprintf("handleVal: mk: %s", mk))
+
+	switch t := v.(type) {
+	case string:
+		log.Info(fmt.Sprintf("ADDING %s=%s to flatConf", k, t))
+		flatConf[mk] = t
+	case bool:
+		log.Info(fmt.Sprintf("ADDING %s=%v to flatConf", k, t))
+		flatConf[mk] = strconv.FormatBool(t)
+	case float64:
+		log.Info(fmt.Sprintf("ADDING %s=%v to flatConf", k, t))
+		flatConf[mk] = strconv.FormatFloat(t, 'f', -1, 64)
+	case map[string]interface{}:
+		log.Info(fmt.Sprintf("FOUND AN OBJECT"))
+
+		for k, v := range t {
+			handleVal(mk, k, v, flatConf)
+		}
+	default:
+		log.Err("I DON'T KNOW!!!!")
+	}
+}
+
+// TODO: merge w/configure version
+func handleSvcConf(env string) {
+	log.Info(fmt.Sprintf("edgex-asc:install:handleSvcConf config is %s", env))
+
+	if env == "" {
+		return
+	}
+
+	var m map[string]interface{}
+	var flatConf = make(map[string]interface{})
+	//flatM = make(map[string]interface{})
+
+	err := json.Unmarshal([]byte(env), &m)
+	if err != nil {
+		log.Err(fmt.Sprintf("edgex-asc:configure:handleSvcConf: failed to unmarshall env; %v", err))
+		return
+	}
+
+	for k, v := range m {
+		handleVal("", k, v, flatConf)
+	}
+
+	path := fmt.Sprintf("%s/config/res/service.env", hooks.SnapDataE)
+
+	f, err := os.OpenFile(path, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0644)
+	if err != nil {
+		log.Err(fmt.Sprintf("edgex-asc:configure:handleSvcConf: can't open %s - %v", path, err))
+		os.Exit(1)
+	}
+
+	defer f.Close()
+
+	log.Info(fmt.Sprintf("edgex-asc:configure:handleSvcConf about write %s", path))
+	for k, v := range flatConf {
+		log.Info(fmt.Sprintf("%s=%v", k, v))
+		_, err := f.WriteString(fmt.Sprintf("export %s=%s\n", hooks.ConfToEnv[k], v))
+		if err != nil {
+			log.Err(fmt.Sprintf("edgex-asc:configure:handleSvcConf: can't open %s - %v", path, err))
+			os.Exit(1)
+		}
+	}
+}
+
+func main() {
+	log, err := syslog.New(syslog.LOG_INFO, "edgex-asc:install")
+	if err != nil {
+		return
+	}
+
+	err = hooks.GetEnvVars()
+	if err != nil {
+		log.Crit(fmt.Sprintf("edgex-asx:install: %v", err))
```

Comment:
there's a few of these sorts of typos

```suggestion
		log.Crit(fmt.Sprintf("edgex-asc:install: %v", err))
```